crates/ruff_linter/src/rules/ruff/rules/explicit_f_string_type_conversion.rs

```diff
@@ -68,84 +66,126 @@ pub(crate) fn explicit_f_string_type_conversion(checker: &Checker, f_string: &as
-        if matches!(
-            arg,
-            Expr::Dict(_) | Expr::Set(_) | Expr::DictComp(_) | Expr::SetComp(_)
-        ) {
-            continue;
-        }
+                // Can't be a conversion otherwise.
+                let [arg] = call.arguments.args.as_ref() else {
+                    continue;
+                };
+                arg
+            }
+        };
 
-        if !checker
-            .semantic()
-            .resolve_builtin_symbol(func)
-            .is_some_and(|builtin| matches!(builtin, "str" | "repr" | "ascii"))
-        {
-            continue;
+        // Suppress lint for starred expressions.
+        if matches!(arg, Expr::Starred(_)) {
+            return;
         }
 
         let mut diagnostic =
             checker.report_diagnostic(ExplicitFStringTypeConversion, expression.range());
         diagnostic.try_set_fix(|| {
-            convert_call_to_conversion_flag(f_string, index, checker.locator(), checker.stylist())
+            convert_call_to_conversion_flag(checker, conversion, element, call, arg)
         });
     }
 }
 
 /// Generate a [`Fix`] to replace an explicit type conversion with a conversion flag.
 fn convert_call_to_conversion_flag(
-    f_string: &ast::FString,
-    index: usize,
-    locator: &Locator,
-    stylist: &Stylist,
+    checker: &Checker,
+    conversion: Conversion,
+    element: &ast::InterpolatedStringElement,
+    call: &ast::ExprCall,
+    arg: &Expr,
 ) -> Result<Fix> {
-    let source_code = locator.slice(f_string);
-    transform_expression(source_code, stylist, |mut expression| {
-        let formatted_string = match_formatted_string(&mut expression)?;
-        // Replace the formatted call expression at `index` with a conversion flag.
-        let formatted_string_expression =
-            match_formatted_string_expression(&mut formatted_string.parts[index])?;
-        let call = match_call_mut(&mut formatted_string_expression.expression)?;
-        let name = match_name(&call.func)?;
-        match name.value {
-            "str" => {
-                formatted_string_expression.conversion = Some("s");
-            }
-            "repr" => {
-                formatted_string_expression.conversion = Some("r");
-            }
-            "ascii" => {
-                formatted_string_expression.conversion = Some("a");
-            }
-            _ => bail!("Unexpected function call: `{:?}`", name.value),
-        }
-        formatted_string_expression.expression = call.args[0].value.clone();
-        Ok(expression)
-    })
-    .map(|output| Fix::safe_edit(Edit::range_replacement(output, f_string.range())))
+    if element
+        .as_interpolation()
+        .is_some_and(|interpolation| interpolation.debug_text.is_some())
+    {
+        anyhow::bail!("Don't support fixing f-string with debug text!");
+    }
+
+    let arg_str = checker.locator().slice(arg);
+    let contains_curly_brace = checker
+        .tokens()
+        .in_range(arg.range())
```

Comment:
I don't think this is correct. E.g we don't want to add a space for an expression like:

```py
1 if b({ "key": "test" }) else 10
```